controllers/wordpressinstance_controller.go

```diff
@@ -50,283 +59,90 @@ type WordpressInstanceReconciler struct {
-	}
-
-	var sb strings.Builder
-
-	data := map[string]interface{}{
-		"UID":       instanceUID,
-		"namespace": instanceNamespace,
+	FillWithDefaults(wp)
+	if err := ApplyResources(ctx, r.Client, *wp); err != nil {
+		return ctrl.Result{RequeueAfter: shortWait}, err
 	}
 
-	tmpl.Execute(&sb, data)
-
-	tmplOutput := sb.String()
-
-	r.Log.V(2).Info("Using yaml", "yaml", tmplOutput)
-
-	// TODO
-	// Return a reconcile error when there's an issue
-	// Set instance owner as the Stack?
-	// Add labels to resources created by individual wordpress instance, and put that
-	//     same label on the instance. For later querying
-	// Maybe put the IP of the load balancer in the wordpress instance CRD
-	// Represent the status of the instance in the CRD?
-	// Use defaults
-	r.Log.V(2).Info("BEFORE extract objects")
-	objects, err := r.ExtractObjects(ctx, &tmplOutput)
+	endpoint, err := GetEndpoint(ctx, r.Client, *wp)
 	if err != nil {
-		r.Log.V(0).Info("Error extracting objects!", "err", err)
+		return ctrl.Result{RequeueAfter: shortWait}, err
 	}
-
-	err = r.createObjects(ctx, objects, i)
-
-	if err != nil {
-		r.Log.V(0).Info("Error creating objects!", "err", err)
+	if endpoint != "" {
+		wp.Status.Endpoint = endpoint
 	}
-
-	return ctrl.Result{}, nil
-}
-
-func (r *WordpressInstanceReconciler) ExtractObjects(ctx context.Context, s *string) ([]*unstructured.Unstructured, error) {
-	// read full output from job by retrieving the logs for the job's pod
-	r.Log.V(2).Info("ENTER extract objects", "ctx", ctx, "string", s)
-	reader := strings.NewReader(*s)
-
-	// decode and process all resources from job output
-	d := yaml.NewYAMLOrJSONDecoder(reader, 4096)
-	var objects []*unstructured.Unstructured
-	for {
-		obj := &unstructured.Unstructured{}
-		if err := d.Decode(&obj); err != nil {
-			if err == io.EOF {
-				// we reached the end of the job output
-				r.Log.V(2).Info("EXIT extract objects because EOF", "objects", objects, "err", err)
-				break
-			}
-			r.Log.V(2).Info("EXIT extract objects because ERROR", "objects", objects, "err", err)
-			return nil, errors.Wrapf(err, "failed to parse output")
-		}
-
-		objects = append(objects, obj)
+	// Update is no-op if there is no change on the object.
+	if err := r.Client.Update(ctx, wp); err != nil {
+		return ctrl.Result{RequeueAfter: shortWait}, err
 	}
-
-	r.Log.V(2).Info("EXIT extract objects", "objects", objects)
-	return objects, nil
+	return ctrl.Result{RequeueAfter: longWait}, nil
 }
 
-func (r *WordpressInstanceReconciler) createObjects(ctx context.Context, objects []*unstructured.Unstructured, i *wordpressv1alpha1.WordpressInstance) error {
-	for _, obj := range objects {
-		// process and create the object that we just decoded
-		if err := r.createOutputObject(ctx, obj, i); err != nil {
-			return err
-		}
+func ApplyResources(ctx context.Context, kube client.Client, wp wordpressv1alpha1.WordpressInstance) error {
+	database := GetMySQLInstance(wp)
```

Comment:
I believe it may be better if all of the resources were applied first, then all errors were collected, and then all errors were returned. That way, resources without errors which may take time to spin up (such as the cluster heh heh) can still make progress even if something else errors out. I believe it'll also make the code a little more friendly since all of the resources will be declared right next to each other in the code.